htdocs/js/jsx/behavioural_feedback_panel.jsx

```diff
@@ -0,0 +1,606 @@
+	  sessionID={that.props.sessionID} type={row.Type}
+        commentID={that.props.commentID} candID={that.props.candID} status={row.QC_status} date={row.Date}
+        commentToggle={that.markCommentToggle.bind(this, index)} fieldname={row.FieldName} author={row.User}
+        onClickClose={that.props.close_thread.bind(this,index)} onClickOpen={that.props.open_thread.bind(this,index)}/>
+      }.bind(this));
+
+	    var table =
+	            <table id ="current_thread_table" className="table table-hover table-primary table-bordered dynamictable">
+        <thead id="current_thread_table_header">
+        {table_headers}
+        </thead>
+        {feedbackRows}
+        </table>
+	
+	
+      return(
+        <div className="panel-collapse collapse in">
+        <div className="panel-body">
+	  {table}
+        </div>
+        </div>
+      )
+    }
+    else{
+	return(
+	            <div className="panel-body">
+
+            There are no threads for this user!
+	            </div>
+
+      )
+    }
+  }
+});
+
+
+var FeedbackPanelRow = React.createClass({
+  getInitialState: function(){
+    return {
+      thread_entries_toggled: false,
+      thread_entries_loaded: false,
+      thread_comment_toggled: false
+    }
+  },
+  componentDidMount: function(){
+    this.loadServerState();
+  },
+  loadServerState: function(){
+    var that = this;
+
+    request = $.ajax({
+      type: "POST",
+      url: "ajax/get_thread_entry_data.php",
+      data:{
+        "callFunc1" : this.props.feedbackID
+      },
+      success:function(data){
+        that.setState({thread_entries_loaded : data});
+      },
+      error: function (xhr, desc, err){
+        console.log(xhr);
+        console.log("Details: " + desc + "\nError:" + err);
+      }
+    });
+  },
+  toggle_entries: function(){
+    this.setState({thread_entries_toggled: !this.state.thread_entries_toggled});
+  },
+  toggle_thread_comment: function(){
+
+    this.setState({thread_comment_toggled : !this.state.thread_comment_toggled});
+
+  },
+  new_thread_entry: function(comment){
+
+    var that = this;
+    var feedbackID = this.props.feedbackID
+    var sessionID = this.props.sessionID;
+    var candID = this.props.candID;
+
+    request = $.ajax({
+      type: "POST",
+      url: "ajax/thread_comment_bvl_feedback.php",
+      data: {"comment" : comment,
+      "feedbackID" : feedbackID,
+      "candID" : candID},
+      success: function (response) {
+        that.loadServerState();
+      },//end of success function
+      error: function (xhr, desc, err){
+        console.log(xhr);
+        console.log("Details: " + desc + "\nError:" + err);
+      }
+    })
+  },
+  render: function() {
+    var d = new Date();
+    var feedbackID = this.props.feedbackID;
+    if (this.state.thread_entries_toggled){
+      var arrow = 'glyphicon glyphicon-chevron-down glyphs';
+      var threadEntries = this.state.thread_entries_loaded.map(function(entry){
```

Comment:
The default value you specified for thread_entries_loaded was false. false.map isn't defined. Should the default be the empty array instead?